moshi/src/main/java/com/squareup/moshi/JsonReader.java

```diff
@@ -214,6 +214,9 @@
   /** True to accept non-spec compliant JSON */
   private boolean lenient = false;
 
+  /** True to throw a {@link JsonDataException} on any attempt to call {@link #skipValue()}. */
+  private boolean skipForbidden = false;
```

Comment:
Should this be `skipAllowed`? I usually find booleans that assert in the positive direction to be easier to reason about.

Think fast: what does `setSkipForbidden(false)` do?